pallets/gear-bank/src/lib.rs

```diff
@@ -175,11 +179,17 @@ pub mod pallet {
         ) -> Result<(), Error<T>> {
             let bank_address = T::BankAddress::get();
 
-            ensure!(
-                CurrencyOf::<T>::free_balance(&bank_address).saturating_add(value)
-                    >= CurrencyOf::<T>::minimum_balance(),
-                Error::InsufficientDeposit
-            );
+            <CurrencyOf<T> as fungible::Inspect<_>>::can_deposit(
+                &bank_address,
+                value,
+                Provenance::Extant,
+            )
+            .into_result()
+            .map_err(|e| match e {
+                DispatchError::Token(TokenError::BelowMinimum) => Error::<T>::InsufficientDeposit,
+                DispatchError::Arithmetic(ArithmeticError::Overflow) => Error::<T>::Overflow,
+                _ => unreachable!("Unexpected error: {e:?}"),
```

Comment:
what cases could cause this branch?

Reply:
In this case only these two are possible. We just need to account for other potential `DispatchError` variants to make it compile.

Reply:
Please no

Other variants of `DispatchError` can be added and this will lead to panic, even if it is impossible now

Please make exhausted match or even better return error and panic in calling code (if absolutely unavoidable)